### ISSUE TITLE
[Accessibility]: Branch/Tag/Commit + Repositories selector: Focus lost on ESC key

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35476]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34417, https://github.com/sourcegraph/sourcegraph/issues/34416

### Problem description

Closing these selectors/popovers should return the focus to the trigger that opened them. It seems that focus is completely lost currently and goes to the first element on the page. This seems to be an issue on quite a few popovers so I have grouped these together.

### Expected behavior

Focus should be returned to the trigger element that opened the selector/popover.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by vovakulikov** [comment]
It was fixed with this PR about improving Popover UI https://github.com/sourcegraph/sourcegraph/pull/43029
